mongodb/activity/queryDocument/activity.go

```diff
@@ -0,0 +1,209 @@
+package queryDocument
+
+import (
+	ctx "context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	mongocon "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+	// "github.com/TIBCOSoftware/flogo-lib/core/activity"
+	// "github.com/TIBCOSoftware/flogo-lib/core/data"
+	// "github.com/TIBCOSoftware/flogo-lib/logger"
+	//"github.com/TIBCOSoftware/flogo-lib/core/data"
+	//"github.com/TIBCOSoftware/flogo-lib/core/data"
+	// Need to remove this after testing with latest API
+	"github.com/project-flogo/core/activity"
+	"github.com/project-flogo/core/support/log"
+
+	//"github.com/project-flogo/core/data/metadata"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// var log = logger.GetLogger("activity-mongodb")
+
+const (
+	connectionProp = "mongoConnection"
+	ivOperation    = "operation"
+	outputProperty = "Output"
+	ivInput        = "input"
+	ovCount        = "count"
+)
+
+func init() {
+	fmt.Println("Entered the init")
+	err := activity.Register(&Activity{}, New)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+// New functioncommon
+func New(ctx1 activity.InitContext) (activity.Activity, error) {
+	fmt.Println("Entered the new method ")
+	settings := &Settings{}
+	//fmt.Println(ctx1.Settings())
+	//fmt.Println("Before Getting the config")
+	settings.Connection, _ = mongocon.GetSharedConfiguration(ctx1.Settings()["mongoConnection"])
```

Comment:
@skothari-tibco 
How do I retrieve activity Settings from activity.InitContext object so as to use metadata.MapToStruct? Is the following line of code the right way instead of the line 49 ?
err := metadata.MapToStruct(ctx.Settings(), settings, true)

Reply:
Yes.

Reply:
@skothari-tibco 
Now I remember why we used 
`settings.Connection, _ = mongocon.GetSharedConfiguration(ctx1.Settings()["mongoConnection"])`
and not `metadata.MapToStruct(ctx.Settings(), settings, true)`
Both our activity and trigger have a settings of type connection.Manager for connection object. If I use `metadata.MapToStruct(ctx.Settings(), settings, true)` in the New method, I run into the following error while running the FE app:

2019-10-29T01:06:27.617+0530 WARN [flogo] - NewDefinition: PANIC Occurred  : reflect.Set: value of type string is not assignable to type connection.Manager

2019-10-29T01:06:27.619+0530 ERROR [flogo] - Failed to create engine instance due to error: error unmarshalling flow: reflect.Set: value of type string is not assignable to type connection.Manager
github.com/project-flogo/core/support/log.(*zapLoggerImpl).Error
	/usr/wi/go/src/github.com/project-flogo/core/support/log/zap.go:43
main.main
	/var/lib/build_server/workspace/bs_302681299/engine/src/engine/main.go:59
runtime.main
	/usr/local/go/src/runtime/proc.go:201

So, I will be sticking to using `settings.Connection, _ = mongocon.GetSharedConfiguration(ctx1.Settings()["mongoConnection"])` in my activities and trigger unless there is a way around this.

Reply:
Well, this error is because the "Connection" in `Settings` is of type connection.Manager .  You can just change it to type string. And then you can add Connection object by using `coerce.ToConnection(settings.MongoConnnection)` to the activity.  Example `https://github.com/skothari-tibco/edge-contrib/blob/mqtt-shared/activity/mqtt/activity.go#L95` . This will also be useful with respect to UI.

Reply:
@skothari-tibco 
Thanks for providing me with a reference. I made the necessary changes and tested the contribution locally, looks like it's holding up. I will check these changes in for both activity and trigger.